### DISCUSSION TITLE
How to call page method from table action/bulk action

### DISCUSSION THREAD

**Discussion by jwhoami** [filamentphp/filament#3818] [Help] [answered]
I have a resource with bulk actions defined, one of the action is like so...

`Tables\Actions\BulkAction::make('toggle-active')
  ->requiresConfirmation()
  ->action(function (Collection $records) {
      callActionDefinedOnPage($records);
    });
  })
  ->label(__('Toggle Active'))
`

How can I refactor the logic related to the "callActionDefinedOnPage()" to a method in the page component for example ListRecords

I tried calling $this->callActionDefinedOnPage($records) but that throws an exception "Using $this when not in object context"

Thanks

**Comment by danharrin** [comment, marked as answer]
Either `->action('callActionDefinedOnPage')` or `->action(fn ($records, $livewire) => $livewire->callActionDefinedOnPage($records))`

**Reply by jwhoami** [reply]
Perfect. +1 for giving more than one option.
Thanks.

**Reply by aliaxonbaig** [reply]
@jwhoami, it would be great if you could share the function logic for mass update, I'm working on the same lines to achieve mass toggle for is_active status on multiple records.

I am able to receive all records in the function, I wonder how to toggle the statuses for the selected records?

thanks!

**Reply by krekas** [reply]
If you can get all records it's just a simple update for each record. Something like:

```php
Tables\Actions\BulkAction::make('Change author')
    ->action(function (Collection $records, array $data): void {
        $records->each->update(['is_active' => YOUR VALUE]]);
    })
    ->deselectRecordsAfterCompletion(),
```

**Reply by aliaxonbaig** [reply]
Thanks for the prompt reply:

I implemented as below:

                    Tables\Actions\BulkAction::make('Toggle Status')
                    ->requiresConfirmation()
                    ->action(function (Collection $records): void {
                    foreach ($records as $record) {
                        $record->is_active = !$record->is_active;
                        $record->save();
                    }
                    Notification::make()
                        ->success()
                        ->title('Status toggled successfully!')
                        ->send();
                    })
                    ->icon('heroicon-m-arrows-right-left')
                    ->deselectRecordsAfterCompletion(),
